grapher/text/MarkdownTextWrap.tsx

```diff
@@ -0,0 +1,527 @@
+            return new IRItalic(
+                parsimmonToTextTokens(node.children, {
+                    ...fontParams,
+                })
+            )
+        } else if (node.type === "textSegments") {
+            return new IRSpan(
+                parsimmonToTextTokens(node.children, {
+                    ...fontParams,
+                })
+            )
+        } else if (node.type === "plainUrl") {
+            return new IRLink(
+                node.href,
+                parsimmonToTextTokens(
+                    [{ type: "text", value: node.href }],
+                    fontParams
+                )
+            )
+        } else if (node.type === "markdownLink") {
+            return new IRLink(
+                node.href,
+                parsimmonToTextTokens(node.children, fontParams)
+            )
+        } else if (node.type === "detailOnDemand") {
+            // TODO: create a IR for this
+            return new IRLink(
+                node.term,
+                parsimmonToTextTokens(node.children, fontParams)
+            )
+        } else {
+            throw new Error(`Unknown node type: ${(node as any).type}`)
+        }
+    })
+}
+
+interface MarkdownTextWrapHTMLProps {
+    isSVG?: false
+    x?: number
+    y?: number
+    svgOptions?: React.SVGProps<SVGTextElement>
+}
+
+interface MarkdownTextWrapSVGProps {
+    isSVG: true
+    x: number
+    y: number
+    svgOptions?: React.SVGProps<SVGTextElement>
+}
+
+type MarkdownTextWrapProps = {
+    text: string
+    maxWidth: number
+    lineHeight?: number
+    fontSize: number
+    fontWeight?: number
+} & (MarkdownTextWrapHTMLProps | MarkdownTextWrapSVGProps)
+
+export class MarkdownTextWrap extends React.Component {
+    props: MarkdownTextWrapProps
+    constructor(props: MarkdownTextWrapProps) {
+        super(props)
+        this.props = props
+    }
+
+    @computed get maxWidth(): number {
+        return this.props.maxWidth ?? Infinity
+    }
+    @computed get lineHeight(): number {
+        return this.props.lineHeight ?? 1.1
+    }
+    @computed get fontSize(): number {
+        return this.props.fontSize ?? 1
+    }
+    @computed get fontWeight(): number | undefined {
+        return this.props.fontWeight
+    }
+    @computed get text(): string {
+        return this.props.text
+    }
+    @computed get ast(): MarkdownRoot["children"] {
+        const result = mdParser.markdown.parse(this.props.text)
+        if (result.status) {
+            return result.value.children
+        }
+        return []
+    }
+
+    @computed get lines(): IRToken[][] {
+        const tokens = parsimmonToTextTokens(this.ast)
+        return splitIntoLines(tokens, this.maxWidth)
+    }
+
+    // @computed get height(): number {
+    //     if (this.lines.length === 0) return 0
+
+    //     return (
+    //         this.lines.reduce((total, line) => total + line.height, 0) +
+    //         this.lineHeight * (this.lines.length - 1)
+    //     )
+    // }
```

Comment:
I think the `height()` method on the original class might be incorrect/non-sensical: https://github.com/owid/owid-grapher/blob/31fc73b21799963e97f908b72b601a85f39979aa/grapher/text/TextWrap.tsx#L132-L139

The `lineHeight` is just used as padding between lines? Or am I missing something? 

Everywhere else `lineHeight` is interpreted as a factor to multiply fontSize by.

Reply:
Yeah seems like a mis-implementation of the line height formula (`font-size * line-height`)

We only need it for SVG so I've just put it in-line.